src/openrct2/peep/Guest.cpp

```diff
@@ -1583,31 +1578,36 @@ bool Guest::DecideAndBuyItem(Ride* ride, int32_t shopItem, money32 price)
         if (itemValue < price)
         {
             itemValue -= price;
-            if (shopItem == SHOP_ITEM_UMBRELLA)
-            {
-                if (climate_is_raining())
-                    goto loc_69B221;
-            }
-
-            itemValue = -itemValue;
-            if (Happiness >= 128)
-                itemValue /= 2;
 
-            if (Happiness >= 180)
-                itemValue /= 2;
-
-            if (itemValue > (static_cast<money16>(scenario_rand() & 0x07)))
+            if (!isRainingAndUmbrella)
             {
-                // "I'm not paying that much for x"
-                PeepThoughtType thought_type = static_cast<PeepThoughtType>(
-                    (shopItem >= 32 ? (PEEP_THOUGHT_TYPE_PHOTO2_MUCH + (shopItem - 32))
-                                    : (PEEP_THOUGHT_TYPE_BALLOON_MUCH + shopItem)));
-                InsertNewThought(thought_type, ride->id);
-                return false;
+                itemValue = -itemValue;
+                if (Happiness >= 128)
+                {
+                    itemValue /= 2;
+                    if (Happiness >= 180)
+                        itemValue /= 2;
+                }
+                if (itemValue > (static_cast<money16>(scenario_rand() & 0x07)))
+                {
+                    // "I'm not paying that much for x"
+                    PeepThoughtType thought_type = static_cast<PeepThoughtType>(
+                        (shopItem >= 32 ? (PEEP_THOUGHT_TYPE_PHOTO2_MUCH + (shopItem - 32))
+                                        : (PEEP_THOUGHT_TYPE_BALLOON_MUCH + shopItem)));
+                    InsertNewThought(thought_type, ride->id);
+                    return false;
+                }
             }
         }
         else
         {
+            // reset itemValue for satisfaction calculation
+            if (gClimateCurrent.Temperature >= 21)
+                itemValue = ShopItems[shopItem].HotValue;
+            else if (gClimateCurrent.Temperature <= 11)
+                itemValue = ShopItems[shopItem].ColdValue;
+            else
+                itemValue = ShopItems[shopItem].BaseValue;
```

Comment:
This is still misplaced, it should be after the happiness calculation, just before line 1631 when it starts the satisfaction calculation

Reply:
You're right. Sorry about that.